wsf_scraping/pipelines.py

```diff
@@ -64,8 +70,8 @@ def check_keywords(self, item, spider_name, base_pdf_path):
         with open(base_pdf_path, 'rb') as f:
             if download_only:
                 os.rename(
-                    ''.join(['/tmp/', item['pdf']]),
-                    ''.join([pdf_result_path, item['pdf']])
+                    os.path.join(tempfile.gettempdir(), item['pdf']),
```

Comment:
Not quite. This always returns the same directory – it’s the directory where temporary files will be created.

Generally prefer using `tempfile.mkdtemp()`, which returns a unique new directory every time.

---

But in general, since the file already exists, you shouldn't be reconstructing the filename! The code duplication makes it very fragile. It would be better to pick a path once with `mkdtemp()`, then pass that around as needed.

Reply:
Concerning `mkdtemp`, is there a real difference, knowing that I delete the file from `/tmp` right after processing it?

Reply:
What if you run two spiders at the same time? It’s a good habit to get into – isolating temporary files like this, so multiple process can’t trample each other – even if you don’t think you need this _right now_.

Reply:
Oh okay, I see now! Thanks, that's much clearer 😄